### PR TITLE
[FIX] calendar: zero count on event repeat raises UserError

### PR DESCRIPTION
**Pull request by jhk-odoo**
Steps to reproduce:
- install calendar
- create entry in Calendar
- make event all day
- select "Repeat / Recurrency"
- select "1 Day" and in Number of Repetitions, / Count select "0"
- save

Previous behavior:
the ui keeps loading and hangs up after 1-2 minutes

Current behavior:
raises an uUserError as intended

opw-2178330

### CODE REVIEW

**Comment by jhk-odoo** [comment]
@jpp-odoo

File: addons/calendar/models/calendar.py [hunk 1]
```diff
@@ -1256,7 +1256,7 @@ def _rrule_serialize(self):
         """
         if self.interval and self.interval < 0:
             raise UserError(_('interval cannot be negative.'))
-        if self.count and self.count <= 0:
+        if self.count <= 0:
```
Comment (jpp-odoo) [inline addons/calendar/models/calendar.py]: Not sure if we need to test if self.count is not None before.
@nim-odoo what do you think ?

Comment (nim-odoo) [reply addons/calendar/models/calendar.py]: What is the value of `self.count` in this case?

Comment (jhk-odoo) [reply addons/calendar/models/calendar.py]: 0

Comment (jhk-odoo) [reply addons/calendar/models/calendar.py]: si it ends up being falsy and does not trigger the initial condition for UserError

Comment (nim-odoo) [reply addons/calendar/models/calendar.py]: By the way, an addtionnal comment would be nice because the reason why this works is not clear.

Comment (jhk-odoo) [reply addons/calendar/models/calendar.py]:
```
# if self.count and ... :
        # ends up being false count is 0
        if hasattr(self, 'count') and self.count <= 0:
```

Comment (nim-odoo) [reply addons/calendar/models/calendar.py]: ok. `self.count <= 0` should be enough, then. It should always be an `int`

Comment (nim-odoo) [reply addons/calendar/models/calendar.py]: side note: from https://github.com/odoo/odoo/blob/3e2ee43fd0aa6a06d5d5ecd1ff669f4a09698330/odoo/fields.py#L1141

**Pull request closed without merging**